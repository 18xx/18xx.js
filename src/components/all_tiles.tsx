import * as Immutable from 'immutable';
import { List } from 'immutable';
import * as React from 'react';
import { ReactElement } from 'react';

import TileDefinition, { TileDefinitionInput } from '../tile_definition';

import * as allTilesJson from '../../config/tiles.json';

const allTiles: List<TileDefinitionInput> =
  Immutable.fromJS(allTilesJson).map(
    (el: any) => el.toJS() as TileDefinitionInput
  );

interface AllTilesProps {
  readonly orientation: string;
}

export default class AllTiles
extends React.Component<AllTilesProps, undefined> {

  public get tiles(): any {
    return allTiles.map(t =>
      <div key={t.num}>
        {new TileDefinition(t, this.props.orientation).allRotations}
      </div>
    );
  }

  public render(): ReactElement<AllTiles> {
    return <div id='all-tiles'>{this.tiles}</div>;
  }
}
