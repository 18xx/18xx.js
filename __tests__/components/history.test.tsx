import { List, Map } from 'immutable';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { allTiles } from '../support/all_tiles';
import { mapDefinition } from '../support/map_definition';

import TileSet, { TileSetDetails } from '../../src/tile_set';

import History, {
  HistoryEntry,
  HistoryProps,
} from '../../src/components/history';

describe('History', () => {
  const tileManifest: Map<string, TileSetDetails> = Map({
    1: { count: 0 },
    7: { count: 1 },
    8: { count: 10 },
    9: {},
  });
  const tileSet: TileSet = new TileSet(allTiles, mapDefinition, tileManifest);

  const placeTile: HistoryEntry = {
    action: 'PLACE_TILE',
    hash: 'tile',
    hex: 'b2',
    id: '7.1',
  };

  const invalidPlaceTile: HistoryEntry = {
    action: 'PLACE_TILE',
    hash: 'tile',
    hex: 'b2',
    id: '58.1',
  };

  const placeToken: HistoryEntry = {
    action: 'PLACE_TOKEN',
    hash: 'token',
    hex: 'a1',
    id: 'B&O',
  };

  const removeToken: HistoryEntry = {
    action: 'REMOVE_TOKEN',
    hash: 'rm-token',
    hex: 'a1',
    id: 'B&O',
  };

  const unknownEntry: HistoryEntry = {
    action: 'UNKNOWN',
    hash: 'unknown',
    hex: 'b1',
    id: '',
  };

  const props: HistoryProps = {
    entries: List([placeTile, placeToken, removeToken, unknownEntry]),
    mapDef: mapDefinition,
    tileSet
  };

  describe('#render', () => {
    it('renders the history', () => {
      const subject: React.ReactElement<History> = (
        <History {...props} />
      );

      expect(renderer.create(subject)).toMatchSnapshot();
    });

    describe('with an unknown tile', () => {
      const badProps: HistoryProps = {
        entries: List([invalidPlaceTile]),
        mapDef: mapDefinition,
        tileSet
      };

      it('throw an error', () => {
        const subject: React.ReactElement<History> = (
          <History {...badProps} />
        );

        expect(() => renderer.create(subject)).toThrowError();
      });
    });
  });
});
