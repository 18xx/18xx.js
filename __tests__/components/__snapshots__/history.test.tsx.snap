// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History #render renders the history 1`] = `
<div
  className="col-sm-3 col-md-2 sidebar"
>
  <h4>
    Recent Moves
  </h4>
  <ol
    reversed={true}
  >
    <li>
      <a
        href="./unknown"
      >
        <span>
          Unknown
        </span>
      </a>
    </li>
    <li>
      <a
        href="./rm-token"
      >
        <span>
          Rm-Tok: 
          B&O
           @ 
          a1
        </span>
      </a>
    </li>
    <li>
      <a
        href="./token"
      >
        <span>
          <svg
            className="token"
            height="40"
            onContextMenu={[Function]}
            opacity={1}
            viewBox="0 0 40 40"
            width="40"
          >
            <circle
              cx={20}
              cy={20}
              fill="#070A3F"
              r={16}
            />
            <rect
              fill="white"
              height={12}
              rx={1.75}
              ry={12}
              width={32}
              x={4}
              y={14}
            />
            <text
              fill="black"
              fontSize={10}
              fontWeight="normal"
              stroke="none"
              textAnchor="middle"
              x={20}
              y={24}
            >
              B&O
            </text>
          </svg>
           
          B&O
           @ 
          a1
        </span>
      </a>
    </li>
    <li>
      <a
        href="./tile"
      >
        <span>
          <span
            className="tile"
          >
            <svg
              height={128}
              viewBox="0 0 110.85125168440814 128"
              width={110.85125168440814}
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                fill="yellow"
                points="0,32 55.42562584220407,0 110.85125168440814,32 110.85125168440814,96 55.42562584220407,128 0,96"
              />
              <path
                d="M27.712812921102035 15.999999999999991 A 32 32 0 0,0 83.13843876330611 16.000000000000014"
                fill="none"
                stroke="black"
                strokeDasharray="1,0"
                strokeWidth={8}
              />
              <text
                fill="black"
                fontSize={8}
                textAnchor="end"
                x={109.85125168440814}
                y={94}
              >
                7.1
              </text>
            </svg>
          </span>
           
          7.1
           @ 
          b2
        </span>
      </a>
    </li>
  </ol>
</div>
`;
